nemo_aligner/models/nlp/gpt/gpt_sft_model.py

```diff
@@ -144,17 +144,31 @@ def finish_validation_step(self):
 
     def generate(
         self,
-        inputs: Tuple[torch.Tensor, torch.Tensor],  # we do not support non-tensor inputs
+        inputs: Union[List[str], Tuple[torch.Tensor, torch.Tensor]],
         length_params: LengthParam,
         sampling_params: SamplingParam = None,
         *,
         strategy: Optional[TextGenerationStrategy] = None,
     ) -> OutputType:
-        """Same as base model generate, except the following
+        """
+        Same as base model generate, except the following:
+
+        1. Apply padding to max length.
+        2. Add a "predictions" key to the output, which is the model output without the prompt.
 
-        1. Going to apply padding to max length
-        2. Going to append "predictions" key which is the model output without the prompt
+        These two additional steps above are only performed for actual generation from the model:
+        if `generate()` is called with `compute_logprob=True` then the base model method is used.
         """
+        if sampling_params is not None and sampling_params.get("compute_logprob", False):
```

Comment:
is compute_logprob=True mutually exclusive with generating new tokens? maybe we should guard the inputs being string in this check too

Reply:
Yeah so as soon as you set `compute_logprob=True` it automatically set other values, like generating a single token: https://github.com/NVIDIA/NeMo/blob/d8ebaa5fb3b7df556b282f3547a605bded5b61de/nemo/collections/nlp/modules/common/text_generation_utils.py#L101

It's supposed to work with tensors as input too (not just strings).

Reply:
ah okay! that makes sense